### ISSUE TITLE
Mapbox streets

### ISSUE THREAD

**Issue by lmoresi** [SciTools/cartopy#1761]
### Description

The interface to access the `mapbox.streets` tiles has been deprecated and attempting to use the current MapboxTiles interface is doomed (returns a 410 error)

#### Code to reproduce

```
import cartopy.io.img_tiles as cimgt
# previous interface which has been deprecated
mapbox_streets = cimgt.MapboxTiles(map_id='streets',  access_token='blah')

```

Below is a potential change to the MapboxTiles class but is not backwards compatible so not something I can suggest a fix for as a PR. 

```
class MapboxTiles(GoogleWTS):
    """
    Implement web tile retrieval from Mapbox.
    For terms of service, see https://www.mapbox.com/tos/.
    """
    def __init__(self, access_token, map_id):
        """
        Set up a new Mapbox tiles instance.
        Access to Mapbox web services requires an access token and a map ID.
        See https://www.mapbox.com/api-documentation/ for details.
        Parameters
        ----------
        access_token
            A valid Mapbox API access token.
        map_id
            An ID for a publicly accessible map (provided by Mapbox).
            This is the map whose tiles will be retrieved through this process.
        """
        self.access_token = access_token
        self.map_id = map_id
        super().__init__()

    def _image_url(self, tile):
        x, y, z = tile
        url = ('https://api.mapbox.com/styles/v1/{id}/tiles/{z}/{x}/{y}'
               '?access_token={token}'.format(z=z, y=y, x=x,
                                              id=self.map_id,
                                              token=self.access_token))
        return url


# new call
mapbox_streets  = MapboxTiles(map_id='mapbox/streets-v11',  access_token='blah')
Mapbox_satellite = MapboxTiles(map_id='mapbox/satellite-v9',  access_token='blah')


```

<details>

### Cartopy version

0.18 and current 0.19.x branch

</details>

**Comment by greglucas** [comment]
If their service is returning a 410 I'm not sure there is much we can do about backwards compatibility.

**Comment by dopplershift** [comment]
I agree with @greglucas

**Comment by lmoresi** [comment]
The Mapbox satellite map still responds to the existing / old url pattern. The new pattern works for both. I am not sure about how the cartopy project would normally deal with a creeping incompatibility of this kind. 

This illustrates the change in the url pattern and the id better matches what Mapbox now describes as their identifier (no longer guaranteed to begin `mapbox.`

```
# OLD code
    
#     def _image_url(self, tile):
#         x, y, z = tile
#         url = ('https://api.mapbox.com/v4/mapbox.{id}/{z}/{x}/{y}.png'
#                '?access_token={token}'.format(z=z, y=y, x=x,
#                                               id=self.map_id,
#                                               token=self.access_token))

# NEW code
    def _image_url(self, tile):
        x, y, z = tile
        url = ('https://api.mapbox.com/styles/v1/{id}/tiles/{z}/{x}/{y}'
               '?access_token={token}'.format(z=z, y=y, x=x,
                                              id=self.map_id,
                                              token=self.access_token))
        return url
```

**Comment by greglucas** [comment]
That looks like it should be a transparent change to Cartopy's users because it is in a private function and takes all of the same inputs. If you're willing to make a PR I think that'd be great!

**Comment by lmoresi** [comment]
Will do ...

**Comment by greglucas** [comment]
Closed by #1763
